### PR TITLE
Fix #1072: Ignore non-actionable vacuum error states

### LINKED ISSUE #1072: Vacuum plugin treats 'unavailable' sensor state as a real error at startup
## Problem

On startup of home-automation, the vacuum plugin reads the initial sensor state of `sensor.valetudo_mellowslimyloris_error` and, if it finds anything other than `""` or `"No error"`, treats it as an active vacuum error and fires an alert.

During a Home Assistant restart (or when the Go service starts before HA has fully connected to the vacuum), the sensor state is `"unavailable"`. The plugin interprets this as a real error and announces:

> "Robot vacuum needs attention: unavailable"

This fires TTS on all configured speakers and an ntfy push notification — a false alarm.

## Timeline from today's startup (2026-05-04)

```
13:41:33 — Vacuum Manager starts
13:41:33 — handleErrorChange sees initial state "unavailable"
13:41:35 — Announcement sent: "Robot vacuum needs attention: unavailable"
           Speakers: kitchen, sitting_room, front_room, kids_bathroom
```

The sensor is currently still `"unavailable"` (last_changed `09:18:08 UTC`), meaning the vacuum integration hasn't connected. This is a normal transient condition, not an error.

## Root Cause

In `manager.go`, `handleErrorChange()` (line ~164) filters out empty strings but does NOT filter `"unavailable"` or `"unknown"`:

```go
value := strings.TrimSpace(newState.State)
if value == "" {
    // Skip transient empty/unknown states.
    return
}
```

The comment even says "transient empty/unknown states" but only checks for empty.

Additionally, `Start()` proactively reads and processes the initial sensor state:

```go
if cur, err := m.haClient.GetState(m.cfg.Vacuum.ErrorSensorID); err == nil && cur != nil {
    m.handleErrorChange(m.cfg.Vacuum.ErrorSensorID, nil, cur)
}
```

This is correct behavior for catching pre-existing errors, but combined with the missing filter it means every startup where HA is still loading will produce a false alarm.

## Other plugins handle this

Several other plugins in the codebase already filter these states:
- `sensorconfig/manager.go:176`: `state.State == "unavailable" || state.State == "unknown"`
- `tv/manager.go:270-279`: Explicitly checks for `"unavailable"` and `"unknown"`
- `environmental/manager.go:1067`: `s == "unknown" || s == "unavailable"`

## Proposed Fix

In `handleErrorChange()`, treat `"unavailable"` and `"unknown"` the same as empty — skip them as transient/non-actionable states:

```go
value := strings.TrimSpace(newState.State)
if value == "" || value == "unavailable" || value == "unknown" {
    return
}
```

This should also clear any tracked error if the sensor transitions from a real error to `"unavailable"` (e.g. vacuum goes offline mid-error), to avoid stale re-announcements.

## Labels

bug

### PR DESCRIPTION
**Pull request by NickBorgers**
Resolves #1072

## Summary
- Treat `unavailable` and `unknown` vacuum error sensor states as non-actionable startup/loading states.
- Clear any tracked vacuum error and repeat cadence when the sensor becomes non-actionable.
- Added unit coverage for startup and stale-repeat behavior.

## Test Plan
- `make unit-tests`
- `make pre-commit`

🤖 Generated by the AI assistant

### CODE REVIEW

**Comment by NickBorgers** [comment]
## Design Review

**Alignment check:** PASS
**Scope check:** PASS

### Conclusion
✅ Approved

The production change is exactly what the issue prescribed (5 lines: extend the guard + clear stale repeat state). Tests cover both paths — startup non-actionable state and mid-error transition to non-actionable — which is the right coverage for this kind of state-machine edge case.

**Comment by NickBorgers** [comment]
## Documentation Review

✅ Approved — bug fix scopes entirely to `vacuum/manager.go` and its tests; no Subscribe changes, no new state variables, no architectural changes, and no flows docs exist for vacuum.

**Comment by NickBorgers** [comment]
## Code Review

### Worth Considering

The state transition docblock at `manager.go:155-159` is now stale — it does not mention `unavailable`/`unknown` (or their behavior: clear + no announcement). One line added there would make the comment match the code.

Also a subtle side-effect: the `value == ""` case previously did a bare `return` (no state mutation); it now also clears `currentError` and `lastAnnouncedAt` as a consequence of being merged into the new branch. In practice an empty sensor value after an error is tracked is unlikely, so this is probably harmless, but it is an undocumented behavior change.

### Conclusion
✅ Approved — logic is correct, mutex and shadow-tracker call ordering is consistent with existing patterns, and both new test cases cover the failure modes from issue #1072.

File: homeautomation-go/internal/plugins/vacuum/manager.go [hunk 1]
```diff
@@ -163,8 +163,13 @@ func (m *Manager) handleErrorChange(entityID string, oldState, newState *ha.Stat
 	}
 
 	value := strings.TrimSpace(newState.State)
-	if value == "" {
-		// Skip transient empty/unknown states.
+	if value == "" || value == "unavailable" || value == "unknown" {
+		// Skip transient/non-actionable states and clear any stale repeat state.
+		m.mu.Lock()
+		m.currentError = ""
+		m.lastAnnouncedAt = time.Time{}
```
Comment (NickBorgers) [inline homeautomation-go/internal/plugins/vacuum/manager.go:171]: Stale docblock — add `any → "unavailable"/"unknown": clear, no announcement` to match the new branch. Also note that `value == ""` used to be a bare `return` (no mutation); it now implicitly clears state too. Probably harmless in practice, but worth a comment if the empty-string semantics are intentionally the same.

**Review by NickBorgers — COMMENTED** [review]
The state transition docblock does not mention the new `unavailable`/`unknown` branch.

**Comment by NickBorgers** [comment]
## Concurrency Review

✅ Approved — small, clean change; the new `unavailable`/`unknown` guard follows the identical lock/unlock + out-of-lock shadow-tracker pattern already established by the `NoErrorValue` branch and `Reset()`, all tests pass with `-race`.

**Comment by NickBorgers** [comment]
## Test Review

✅ Approved — single-plugin guard fix with correct mutex handling, zero anti-patterns, 13/13 tests passing including new table-driven coverage for the added `"unavailable"`/`"unknown"` paths.

**Comment by NickBorgers** [comment]
## 🚦 Merge Decision

**Decision: 🟢 GO**

**Findings addressed:**

- **Code Review — "Worth Considering" #1: Stale docblock at `manager.go:155-159`**
  Fixed: Added a `""`/`"unavailable"`/`"unknown"` entry to the state transition table so the comment matches the code.

- **Code Review — "Worth Considering" #2: `value == ""` now clears state instead of bare `return`**
  Accepted as-is. The new behavior (clear `currentError`/`lastAnnouncedAt` on empty string) is strictly safer than the prior bare `return` — it prevents stale repeat state if an empty value ever follows a real error. The reviewer acknowledged it as "probably harmless"; the new behavior is in fact an improvement and is consistent with the `unavailable`/`unknown` handling added in this PR.

- **No merge conflicts:** `origin/main` is an ancestor of `HEAD`.

- **Cross-PR check:** No conflicts with recently merged PRs. The `AlertNotifier` refactor (#1063) and urgency collapse (#1052) are already reflected in the vacuum plugin's current code; this PR makes no changes to notification logic, only to the sensor state guard.

Ready for merge to main and production deployment.

Applied fixes: Updated `handleErrorChange` docblock to document the new non-actionable state transition (commit `8836f56`).

**Comment by NickBorgers** [comment]
Automated conflict resolution skipped: PR #1073 is no longer conflicting. Current mergeStateStatus is CLEAN and mergeable is MERGEABLE, so there is no conflict to resolve.
